synapse/storage/databases/main/event_federation.py

```diff
@@ -140,6 +141,15 @@ def __init__(
 
         self._clock.looping_call(self._get_stats_for_federation_staging, 30 * 1000)
 
+        if isinstance(self.database_engine, PostgresEngine):
+            self.db_pool.updates.register_background_validate_constraint_and_delete_rows(
+                update_name="event_forward_extremities_constraint_update",
+                table="event_forward_extremities",
+                constraint_name="event_forward_extremities_event_id",
+                constraint=ForeignKeyConstraint("events", [("event_id", "event_id")]),
+                unique_columns=("event_id", "room_id"),
+            )
```

Comment:
Why is it most relevant to put the registration in this file vs others? (just curious if any reasoning)

Reply:
We just tend to do most operations on `event_forward_extremities` in this file :shrug: